.github/workflows/snapshot-test.yml

```diff
@@ -0,0 +1,40 @@
+name: Zeebe SNAPSHOT test
+on:
+  workflow_call: { }
+  schedule:
+    - cron: "0 3 * * *"
+jobs:
+  test-against-zeebe-snapshot:
+    name: Run tests against Zeebe SNAPSHOT version
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Java environment
+        uses: actions/setup-java@v3.4.1
+        with:
+          distribution: temurin
+          java-version: 17
+          cache: maven
+
+      - name: Build with Zeebe SNAPSHOT version
+        run: |
+          CURRENT_ZEEBE_VERSION=$(mvn help:evaluate -Dexpression=dependency.zeebe.version -DforceStdout -q)
+          if grep -q "alpha" <<< "$CURRENT_ZEEBE_VERSION"; then
+            NEW_ZEEBE_VERSION=$(echo $CURRENT_ZEEBE_VERSION | sed "s/alpha.*/SNAPSHOT/")
+          else
+            NEW_ZEEBE_VERSION=$(echo $CURRENT_ZEEBE_VERSION | awk -F. '/[0-9]+\./{$2++;print}' OFS=.)
+          fi
+          echo $NEW_ZEEBE_VERSION
+          mvn -B -U -pl "-:zeebe-process-test-qa-testcontainers" -P !localBuild "-Dsurefire.rerunFailingTestsCount=5" -Ddependency.zeebe.version=$NEW_ZEEBE_VERSION clean install
```

Comment:
Too much command line magic for me :dizzy_face:

Reply:
It was a lot of googling for me too 😅 

Long story short what it does:
1. Read the `dependency.zeebe.version` from the pom file and store it in the `CURRENT_ZEEBE_VERSION` variable.
2. If the `CURRENT_ZEEBE_VERSION` is an alpha version it creates the `NEW_ZEEBE_VERSION` variable. This gets the value of the `CURRENT_ZEEBE_VERSION`, but replaces alpha and everything that comes after this with `SNAPSHOT`. E.g. `8.1.0-alpha4` becomes `8.1.0-SNAPSHOT`.
3. If it's not an alpha version we do some actual magic I stole from a StackOverflow post 🪄 It takes the `CURRENT_ZEEBE_VERSION` and from what I understand it groups the version number into 3 sections each divided by a dot. So that would be the major, minor and patch version. `{$2++;print}` tells us to get the 2nd group (the minor version) and increment it with 1. I have no clue what `OFS=.` does.
4. Afterwards we just run the tests using maven overriding the zeebe version with `NEW_ZEEBE_VERSION` (`-Ddependency.zeebe.version=$NEW_ZEEBE_VERSION`)

Reply:
Thank you for explaining :tophat: :magic_wand: